src/pluginrc/saltplugins.rc

```diff
@@ -1,17 +1,16 @@
 #!/bin/bash
 
 function check_packages() {
-    RPMS=$1
-    for pkg in $RPMS; do
-	if ! rpm -q $pkg &>/dev/null; then
-	    echo "ERROR: Salt package \"$pkg\" is not installed"
-	    echo
-	    exit 111
+    for pkg in "$@"; do
+	if ! validate_rpm_pkg $pkg; then
+	    exit 111 
```

Comment:
This confuses me a little bit. Since it is possible to pass several RPM pkgs, does it make sense to exit once one is not valid? Are we then ignoring the status of the rest?

Reply:
According to the way as the function call is used it seems we need to check all listed packages before proceeding all the rest plugin.
But this part:
 RPMS=$1
 for pkg in $RPMS; do
Always takes just 1st package from the list, but not the rest.

Reply:
My take: it is correct to replace `RPMS=$1` with `RPMS=$@` (which was basically done here) because all packages passed should be checked. Looking at how this function is named and used, the previous implementation seems to be a bug.

The question now is how to handle the case when not all checks are positive. Should we return on the first missing package or keep looping through all and return a list of invalid packages? The latter sounds better to me because the goal is to gather information and we potentially lose information if we exit early. So instead of exiting we could add `$pkg` to a `$bad_rpms` and later check if that is empty to trigger the `exit 111`.

Reply:
Maybe we can have it in a second PR, as someone might (rightfully) think that it works like this and uses this out of context.